index.js

```diff
@@ -87,6 +107,22 @@ MarketplaceClient.prototype.validatePackage = function(packagePath) {
         }, function(error, response, body) {
           if (error) {
             reject(error);
+          } else if (response.statusCode == 202) {
+            // Checks the validation's status every five seconds, if it's not yet processed
+            var poller = function() {
+              setTimeout(function() {
+                self.getValidation(body.id).then(function(validation) {
+                  if (validation.processed) {
+                    resolve(validation);
+                  } else {
+                    poller();
+                  }
+                }, function() {
```

Comment:
@tofumatt yes, this promise will not resolve until it gets a 202 back, which would result in an infinite loop if there is an error code in the 4XX range.
Is there any good way to cancel a promise, since I do not see any way other than the user stopping node's execution for the polling to be cancelled with this implementation.